src/main.h

```diff
@@ -77,17 +77,20 @@ static const int MAX_SCRIPTCHECK_THREADS = 16;
 /** -par default (number of script-checking threads, 0 = auto) */
 static const int DEFAULT_SCRIPTCHECK_THREADS = 0;
 /** Number of blocks that can be requested at any given time from a single peer. */
-static const int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 16;
+//static const int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 16;
+static int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 1; // BU request manager just needs one per peer
```

Comment:
The comment should be " BU request manager just needs one per peer to
start with"    we want to start with a low number in case we request
from a slow peer right at the beginning.  as the blocks start coming in
the value gets adjusted upward as needed.

09/2016 3:28 AM, gandrewstone wrote:
>
> In src/main.h
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/79#discussion_r77450097>:
>
> > @@ -77,17 +77,20 @@ static const int MAX_SCRIPTCHECK_THREADS = 16;
> >  /** -par default (number of script-checking threads, 0 = auto) */
> >  static const int DEFAULT_SCRIPTCHECK_THREADS = 0;
> >  /** Number of blocks that can be requested at any given time from a single peer. */
> > -static const int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 16;
> > +//static const int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 16;
> > +static int MAX_BLOCKS_IN_TRANSIT_PER_PEER = 1; // BU request manager just needs one per peer
>
> Comment is misleading, you are adjusting this dynamically based on
> block DL perf, right? If so I will fix during merge.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/79/files/efaaa2474cc017ebc1c84fa3748b50c293bff5c7#r77450097>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMRF0L-GQJD1uxSNLmOl5gdPABhDatGLks5qmp1MgaJpZM4JokbP>.
>